src/array.jl

```diff
@@ -482,27 +482,58 @@ end
 
 # optimize reshape to return a CuArray
 
-function Base.reinterpret(::Type{T}, a::CuArray{S,N}) where {T,S,N}
-  if N == 0 && sizeof(T) == sizeof(S)
```

Comment:
BTW, I think `sizeof(T) == sizeof(S)` was a typo? This is now replaced with `sizeof(T) != sizeof(S)`. I added a test `@test collect(reinterpret(Int32, CUDA.fill(1f0)))[] == reinterpret(Int32, 1f0)` for this.